lib/exercises.rb

```diff
@@ -2,19 +2,75 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(m * n), where m is the number of words in the strings array and n is the (worst case?) length of the words
+# Space Complexity: O(m), could have been O(m + n) but n is at most 26 or 52
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  return [] if strings.length == 0
+  
+  anagram_hash = {}
+  
+  strings.each do |word|
+    letters_hash = {}
+    
+    word.each_char do |char|
+      if letters_hash[char]
+        letters_hash[char] += 1
+      else
+        letters_hash[char] = 1
+      end
+    end
+    
+    if anagram_hash[letters_hash]
+      anagram_hash[letters_hash] << word
+    else
+      anagram_hash[letters_hash] = [word]
+    end
+  end
+  
+  return anagram_hash.values
 end
 
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2), where n is the number of numbers in list (worst case scenario)
+# Space Complexity: O(n)
 def top_k_frequent_elements(list, k)
```

Comment:
Looking at the 2nd loop it seems that your time complexity would be O(k * n) where n is the number of distinct numbers in the list.